### PR TITLE
Fix an invalid memory reference.

### PR DESCRIPTION
**Pull request by drwells**
There should only be one thing in this array.

### IBAMR Pull Request Checklist
- [x] Does the test suite pass?
- [x] Was clang-format run?
- [x] Were relevant issues cited? Please remember to add `Fixes #12345` to close
      the issue automatically if we are fixing the problem.
- [x] Is this a change others will want to know about? If so, then has a
      changelog entry been added?
- [x] If new data structures have been added to a class then have they been set
      up appropriately for restarts? If so, ensure that the restart version
      number is incremented.
- [x] Does this change include a bug fix or new functionality? If so, a new test
      or tests should be added. New tests should run quickly (less than a minute
      in release mode). If possible, an older test should gain a new option so
      that we do not need to compile more test executables.
- [x] Did you (if your account has permission to do so) set relevant labels on
      GitHub for the pull request?

### CODE REVIEW

File: tests/adv_diff/adv_diff_regridding.cpp [hunk 1]
```diff
@@ -109,15 +109,15 @@ main(int argc, char* argv[])
 
         // Create boundary condition specification objects (when necessary).
         const IntVector<NDIM>& periodic_shift = grid_geometry->getPeriodicShift();
-        std::vector<RobinBcCoefStrategy<NDIM>*> Q_bc_coefs(1);
+        std::vector<RobinBcCoefStrategy<NDIM>*> Q_bc_coefs;
```
[the remark this answers is not in the document]
Comment (boyceg) [reply tests/adv_diff/adv_diff_regridding.cpp:112]: And then only set it if the BCs are not all periodic?

**Review by boyceg — APPROVED** [review]

File: tests/adv_diff/adv_diff_regridding.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (drwells) [reply tests/adv_diff/adv_diff_regridding.cpp:112]: We could do that, but I prefer the present version since this is the same check we use everywhere else.
